### PR TITLE
fix(gatsby-source-contentful): Prevent TypeError if many-to-one reference already exists

### PR DESCRIPTION
**Pull request by froddd**
## Description
When creating nodes, linkages to foreign references are created. If a foreign reference is of type many-to-one, the reference is recorded as a string ([normalize.js:330](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-source-contentful/src/normalize.js#L330)). When creating the reverse linkage ([normalize.js:345](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-source-contentful/src/normalize.js#L345)), `createContentTypeNodes()` attempts to push to this foreign reference. As it is, in this case, a string, this results in a TypeError.

The above can be tested using the following content type schema:
```
Blog Post:
- title: string
- slug: slug
- author: single Linked entry (Author)
Author:
- name: string
- favouritePost: single Linked entry (Blog Post)
```
With one Blog post linking to one Author which itself links to the Blog Post, the node creation will fail with the following error:

```
 TypeError: entryItemFields[foreignReference.name].push is not a function

      343 |           const existingReference = entryItemFields[foreignReference.name]
      344 |           if (existingReference) {
    > 345 |             entryItemFields[foreignReference.name].push(
          |                                                    ^
      346 |               mId(foreignReference.id)
      347 |             )
      348 |           } else {

      at push (packages/gatsby-source-contentful/src/normalize.js:345:52)
          at Array.forEach (<anonymous>)
``` 
## Fix

I've added the above simple schema to the test data for `normalize.js`, and fixed by adding a `Array.isArray()` check (this approach seems favoured earlier: [normalize.js:298](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-source-contentful/src/normalize.js#L298)), for consistency, rather than checking that the reference is _not_ a string.

Snapshots have been updated.

## Related Issues

Related to #3064 (possibly?)

### CODE REVIEW

**Comment by wardpeet** [comment]
Hey @froddd!

Thanks for this PR! We really appreciate you taking the time to open it. This PR looks good but to have more confident in moving this forward I would love to know how to test this outside of jest.

In the interest of moving this along most effectively, we strongly encourage our contributors to provide an example project which we can then use to validate the changes in this PR. This will allow us to quickly and easily ensure that this does indeed address the purported feature or fix in this PR!

An example project:

can be run locally or credentials have been provided
Feel free to message on Twitter, Discord, etc.
can be used to validate this functionality
In other words, the necessary functionality has been enabled/disabled in the project for us to easily validate these > changes>
If you're able to provide an example that we can use to test these changes we can more quickly get this merged and released. If you're unable--we will prioritize as best as we can and test when we're able to do so.

Thanks for this PR and thanks for using Gatsby!

💜

**Comment by froddd** [comment]
Sure thing -- example repo is a straight clone of the Gatsby Contentful Starter project with one migration added: https://github.com/froddd/gatsby-contentful-starter

Install as normal and link to your contentful space:
```
$ yarn install
$ yarn run setup
```

Add an `Author` content type with a `favouritePost` field which links to a single `blogPost` content type -- you can run the migration I've provided for this:
```
$ contentful space migration -s <your space id> --mt <your management token> contentful/migrations/000-create-author-content-type.js
```

At this point, if you simply run the dev site locally (`yarn run dev`), everything should be fine.

Add an Author which has one of the posts as favourite and publish it:
Screenshot 2019-09-09 at 11 08 51

Now clear the cache and re-run the local dev environment:
```
$ rm -rf .cache && rm -rf public && yarn run dev
```

You will get the following error:
```
error Plugin gatsby-source-contentful returned an error


  TypeError: entryItemFields[foreignReference.name].push is not a function

  - normalize.js:331 foreignReferences.forEach.foreignReference
    [gatsby-contentful-starter]/[gatsby-source-contentful]/normalize.js:331:52

[stack trace truncated]
```

**Comment by froddd** [comment]
Is there a way to link to this PR's `gatsby-source-contentful` package in the example project to see the fixed behaviour? 🤔

**Comment by froddd** [comment]
When linking locally this now produces another error:

```
error UNHANDLED REJECTION


  Error: ContentfulAsset.fixed provided incorrect OutputType: 'ContentfulFixed'

  - TypeMapper.js:294 TypeMapper.convertOutputFieldConfig
    [gatsby-contentful-starter]/[graphql-compose]/lib/TypeMapper.js:294:15

  - configAsThunk.js:19 resolveOutputConfigAsThunk
    [gatsby-contentful-starter]/[graphql-compose]/lib/utils/configAsThunk.js:19:41

  - ObjectTypeComposer.js:300 ObjectTypeComposer.getFieldConfig
    [gatsby-contentful-starter]/[graphql-compose]/lib/ObjectTypeComposer.js:300:58

  - toInputObjectType.js:44 fieldNames.forEach.fieldName
    [gatsby-contentful-starter]/[graphql-compose]/lib/utils/toInputObjectType.js:44:19

  - Array.forEach
```

I'll do more looking into this.

**Comment by froddd** [comment]
Looks like that extra error could be due to this: https://github.com/gatsbyjs/gatsby/issues/16455

**Comment by froddd** [comment]
I've created a better, more stripped-down example to validate the bugfix: https://github.com/froddd/gatsby-contentful-test

**Comment by wardpeet** [comment]
Published in `gatsby-source-contentful@2.1.35`
